src/System.Device.Gpio/System/Device/Gpio/GpioController.cs

```diff
@@ -201,12 +201,12 @@ public void Write(int pinNumber, PinValue value)
             int logicalPinNumber = GetLogicalPinNumber(pinNumber);
             if (!_openPins.Contains(logicalPinNumber))
             {
-                throw new InvalidOperationException("Can not write to a pin that is not open.");
+                throw new InvalidOperationException($"Can not write to pin {logicalPinNumber} because is not open.");
             }
 
             if (_driver.GetPinMode(logicalPinNumber) != PinMode.Output)
             {
-                throw new InvalidOperationException("Can not write to a pin that is not set to Output mode.");
+                throw new InvalidOperationException($"Can not write to pin {logicalPinNumber} because is not set to Output mode.");
```

Comment:
```suggestion
                throw new InvalidOperationException($"Can not write to pin {logicalPinNumber} because it is not set to Output mode.");
```